src/pydicom/dataset.py

```diff
@@ -2055,8 +2056,9 @@ def decompress(
           *Pixel Data* element will be set to ``False``.
         * Any :dcm:`image pixel<part03/sect_C.7.6.3.html>` module elements may be
           modified as required to match the uncompressed *Pixel Data*.
-        * If `generate_instance_uid` is ``True`` (default) then a new (0008,0018) *SOP
-          Instance UID* value will be generated.
+        * If the color space has been converted from YCbCr to RGB then a new
+          (0008,0018) *SOP Instance UID* value will be generated unless
+          `generate_instance_uid` is ``False``.
 
         .. versionchanged:: 3.0
```

Comment:
Nitpick here, but version changed should be updated to 3.1 (or some 3.0.x if releasing a patch version, I guess).

Reply:
I saw it as more of a fix than a change, the behaviour is similar, the default effect is the same, only it now affects the correct subset of datasets rather than all.

Reply:
I can see it both ways - as a change in current behavior, or a fix to the correct behavior.  It's okay we can leave as is, it is unlikely to affect very many people anyway.